### DISCUSSION TITLE
How to set trigger for complicated schedule

### DISCUSSION THREAD

**Discussion by tolsen64** [dahall/TaskScheduler#916] [Q&A] [answered]
How would I set up a task to run every 15 minutes, between 6am and 11pm, Monday - Friday, indefinitely?  Is that even possible?

**Comment by dahall** [comment, marked as answer]
I think this trigger does what you need:
```cs
WeeklyTrigger wTrigger = new WeeklyTrigger();
// Start the trigger at 6:00 a.m. local time
wTrigger.StartBoundary = DateTime.Today + TimeSpan.FromHours(6);
// Sets to run on weekdays
wTrigger.DaysOfWeek = DaysOfTheWeek.Monday | DaysOfTheWeek.Tuesday | DaysOfTheWeek.Wednesday | DaysOfTheWeek.Thursday | DaysOfTheWeek.Friday;
// Run it every week
wTrigger.WeeksInterval = 1;
// Repeat every 15 minutes for 17 consecutive hours (6am-11pm == 17 hours)
wTrigger.SetRepetition(TimeSpan.FromMinutes(15), TimeSpan.FromHours(17), false);
```

**Reply by tolsen64** [reply]
Thanks! That helped alot.  However, some tasks only need to run once per day at the start time.  The only way I could figure out to do it was this:


    RepetitionPattern repetitionPattern;
    if (jobDefinition.IntervalMinutes > 0)
        repetitionPattern = new(TimeSpan.FromMinutes(jobDefinition.IntervalMinutes), tsEndTime.Subtract(tsStartTime), false);
    else
        repetitionPattern = new(TimeSpan.FromMinutes(1), TimeSpan.FromMinutes(1), false);


Is there a better way?

**Comment by dahall** [comment]
Yes. Omit any reference to the `RepetitionPattern`. By default, the `WeeklyTrigger` will run just once on the specified days at the `StartBoundary` time. In my example, omit the call to `SetRepeition`.
